gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -290,7 +290,7 @@ def prop_validate_definition!(name, cls, rules, type)
   def prop_defined(name, cls, rules={})
     cls = T::Utils.resolve_alias(cls)
 
-    if T::Utils::Nilable.is_union_with_nilclass(cls)
+    if T::Utils::Nilable.is_union_with_nilclass(cls) || (rules.key?(:default) && rules[:default].nil?)
```

Comment:
Nit: Should we turn this into a `prop_nilable?` method?

Reply:
Sure, should it be private in this file?

Reply:
Sounds reasonable!